src/test/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationServiceTest_Generate_Shift.kt

```diff
@@ -139,16 +141,16 @@ internal class NotificationServiceTest_Generate_Shift {
             every { csrClient.getShiftNotifications(any(), any()) } returns listOf(dto1, dto2, dto3)
             every { csrClient.getShiftTaskNotifications(any(), any()) } returns listOf()
 
-            every { shiftNotificationRepository.countAllByQuantumIdAndShiftDateAndShiftTypeAndShiftModified(quantumId, any(), shiftType, today) } returns 0
+            every { shiftNotificationRepository.countAllByQuantumIdIgnoreCaseAndShiftDateAndShiftTypeIgnoreCaseAndShiftModified(quantumId, any(), shiftType, today) } returns 0
 
-            val slot = slot<Collection<ShiftNotification>>()
-            every { shiftNotificationRepository.saveAll(capture(slot)) } returns  listOf()
+            val results = mutableListOf<Collection<ShiftNotification>>()
```

Comment:
Because we are saving more than once, we need to store the captures in a list which we can then refer to each element.